app/gui2/src/composables/navigator.ts

```diff
@@ -288,13 +288,18 @@ export function useNavigator(
     keyboard,
     WHEEL_CAPTURE_DURATION_MS,
     (e, inputType) => {
+      const clientPos = eventScreenPos(e)
+      const scenePos = clientToScenePos(clientPos)
       if (inputType === 'trackpad') {
         // OS X trackpad events provide usable rate-of-change information.
         updateScale((oldValue: number) => oldValue * Math.exp(-e.deltaY / 100))
       } else {
         // Mouse wheel rate information is unreliable. We just step in the direction of the sign.
         stepZoom(-Math.sign(e.deltaY))
       }
+      const scenePos1 = clientToScenePos(clientPos)
+      targetCenter.value = center.value.add(scenePos.sub(scenePos1))
+      center.skip()
```

Comment:
1. Add a comment what we're doing here - it may be not obvious from just add/sub operations
2. nit: I would rename scenePos to `scenePos0`